src/app.js

```diff
@@ -165,26 +173,65 @@ const telemetryConfig = {
 
 telemetry.init(telemetryConfig)
 
-// function to update the config
-function updateConfig (configString) {
-  configUtil.setConfig('CHANNEL_FILTER_QUERY_STRING', configString)
-}
-
-// function to generate the search string
-function getFilterConfig () {
+// function to generate the search filter and return JSON Object
+function getMetaFilterConfig () {
   LOG.info(utilsService.getLoggerData({}, 'INFO',
     filename, 'getFilterConfig', 'environment info', process.env))
   var allowedChannels = whiteListedChannelList ? whiteListedChannelList.split(',') : []
   var blackListedChannels = blackListedChannelList ? blackListedChannelList.split(',') : []
+  var allowedFramework = whitelistedFrameworkList ? whitelistedFrameworkList.split(',') : []
+  var blackListedFramework = blacklistedFrameworkList ? blacklistedFrameworkList.split(',') : []
+  var allowedMimetype = whitelistedMimeTypeList ? whitelistedMimeTypeList.split(',') : []
+  var blackListedMimetype = blacklistedMimeTypeList ? blacklistedMimeTypeList.split(',') : []
+  var allowedContenttype = whitelistedContentTypeList ? whitelistedContentTypeList.split(',') : []
+  var blackListedContenttype = blacklistedContentTypeList ? blacklistedContentTypeList.split(',') : []
+  var allowedResourcetype = whitelistedResourceTypeList ? whitelistedResourceTypeList.split(',') : []
+  var blackListedResourcetype = blacklistedResourceTypeList ? blacklistedResourceTypeList.split(',') : []
+
+  var channelConf = generateConfigString(allowedChannels, blackListedChannels)
+  var frameworkConf = generateConfigString(allowedFramework, blackListedFramework)
+  var mimeTypeConf = generateConfigString(allowedMimetype, blackListedMimetype)
+  var contentTypeConf = generateConfigString(allowedContenttype, blackListedContenttype)
+  var resourceTypeConf = generateConfigString(allowedResourcetype, blackListedResourcetype)
+
```

Comment:
Refactor the code and push it in a new file. app.js should call that file

Reply:
Done, removed the lines and created new file